digest.go

```diff
@@ -62,6 +62,7 @@ func Digest(chal *Challenge, o Options) (*Credentials, error) {
 	var h hash.Hash
 	switch cred.Algorithm {
 	case "", "MD5":
+		cred.Algorithm = "MD5"
```

Comment:
I'm not sure we need this.

Reply:
Yes, it is not important 
I removed but left the test where a challenge did not return a Algorithm (it is my case)